pyteal/ast/abi/tuple_test.py

```diff
@@ -371,6 +371,15 @@ class IndexTest(NamedTuple):
                 encoded, start_index=pt.ExtractUint16(encoded, pt.Int(0))
             ),
         ),
+        IndexTest(
```

Comment:
I tried this test case against `master` (and also the graviton roundtrip test that I proposed) and both tests passed. I'd like to get a better understanding of why that is. For the graviton roundtrip test, I'm guess the test is passing because its decoding the entire tuple as a posed to a portion of it.

Reply:
good point on reverse testing on `master`, I am also trying to replicate things on `master` and will get back to you what I have.

Reply:
ok, my first find: this is INDEED an insufficient test: for we are taking from byte, rather than a static array (with static type). So the compilation will use `getbyte`, which makes no difference with or without length.

I propose we should introduce another type spec (like static bytes) for this testcase.

Reply:
made a change in https://github.com/algorand/pyteal/pull/663/commits/7bc748d4c7c38adbc139443726c65637aa2449ad, and it should be failing on master